### PR TITLE
Feature/dashboard testing

### PR DESCRIPTION
**Pull request by edlsto**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds testing for the dashboard component. All tests are passing

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/dashboard/__tests__/dashboard.test.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        time="4"
+        articles="5"
+        avgArticles="6"
+        words="7"
+        avgWords="8"
+      />,
+    );
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (iEv0lv3) [inline app/javascript/dashboard/__tests__/dashboard.test.jsx:21]: Great testing! It hits all of the necessary statistics for display, well done.

File: app/javascript/dashboard/dashboard.jsx [hunk 2]
```diff
@@ -4,16 +4,23 @@ import displayClick from '../../assets/images/display_click.png';
...
   render() {
-    /* eslint-disable */ const { visits, visit, comments, time, articles, avgArticles, words, avgWords } = this.props;
-console.log(avgArticles)
+    /* eslint-disable */ const {
+      visits,
+      visit,
+      comments,
+      time,
+      articles,
+      avgArticles,
```
Comment (iEv0lv3) [inline app/javascript/dashboard/dashboard.jsx:20]: I assume using camel case here doesn't break anything?

File: app/javascript/dashboard/dashboard.jsx [hunk 3]
```diff
@@ -4,16 +4,23 @@ import displayClick from '../../assets/images/display_click.png';
...
+      words,
+      avgWords,
```
Comment (iEv0lv3) [inline app/javascript/dashboard/dashboard.jsx:22]: Same for this line, I assume using camel case doesn't break anything?

**Review by iEv0lv3 — COMMENTED** [review]
Overall this looks great. Did you run RSpec to check if there were any conflicts or broken tests after making these changes?

File: app/javascript/dashboard/__tests__/dashboard.test.jsx (hunk 1 above)
Comment (edlsto) [reply app/javascript/dashboard/__tests__/dashboard.test.jsx:21]: Thank you!

File: app/javascript/dashboard/dashboard.jsx (hunk 2 above)
Comment (edlsto) [reply app/javascript/dashboard/dashboard.jsx:20]: The casing was an issue with the HTML data attributes. It should be fine within Preact.

File: app/javascript/dashboard/dashboard.jsx (hunk 3 above)
Comment (edlsto) [reply app/javascript/dashboard/dashboard.jsx:22]: Yes, it was only the HTML data attributes where we were having the casing issues.

File: app/javascript/dashboard/__tests__/dashboard.test.jsx (hunk 1 above)
Comment (edlsto) [reply app/javascript/dashboard/__tests__/dashboard.test.jsx:21]: I did run RSpec, and did not get any errors related to these files.

File: app/javascript/dashboard/dashboard.jsx (hunk 2 above)
Comment (iEv0lv3) [reply app/javascript/dashboard/dashboard.jsx:20]: Excellent, thank you for the clarification.

**Comment by iEv0lv3** [comment]
I ran the full RSpec suite locally and the same 2 tests are failing @edlsto. We inherited those and they don't affect our work, so I'm going to merge this PR. I'll add a card to our project board to look at those failing tests.

**Pull request merged**